dev/PushNotifications/PushNotificationsLongRunningTask/ForegroundSinkManager.cpp

```diff
@@ -0,0 +1,26 @@
+﻿#pragma once
+
+#include "pch.h"
+
+#include <NotificationsLongRunningProcess_h.h>
+#include <ForegroundSinkManager.h>
+
+void ForegroundSinkManager::Add(std::wstring processName, IWpnForegroundSink* sink)
+{
+    m_foregroundMap[processName] = sink;
```

Comment:
None of these operations have locks or are threadsafe.

Reply:
Added lock to ForegroundSinkManager.